sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -131,7 +131,8 @@ object Cast {
     case (from: DecimalType, to: NumericType) if from.isTighterThan(to) => true
     case (f, t) if legalNumericPrecedence(f, t) => true
     case (DateType, TimestampType) => true
-    case (_, StringType) => true
+    case (_: AtomicType, StringType) => true
+    case (_: CalendarIntervalType, StringType) => true
```

Comment:
How about `UserDefinedType`?

Reply:
I don't think it is valid up-cast.
Even in `Cast`, Spark only allows casting to the same udt.
```
case udt: UserDefinedType[_]
          if udt.userClass == from.asInstanceOf[UserDefinedType[_]].userClass =>
          identity[Any]
case _: UserDefinedType[_] =>
          throw new SparkException(s"Cannot cast $from to $to.")
```

Reply:
Yup, I also think the case is not valid. So, my question is just that the cast is accepted before this pr? If so, it might be better to add tests for that, too.